packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -829,10 +829,16 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
                     requestQueue: await this.getRequestQueue(),
                 });
             },
-            sendRequest: async (overrideOptions?: GotOptionsInit) => {
+            sendRequest: async (overrideOptions?: OptionsInit) => {
+                const cookieJar = session ? {
+                    getCookieString: (url: string) => session!.getCookieString(url),
+                    setCookie: (rawCookie: string, url: string) => session!.setCookie(rawCookie, url),
+                    ...overrideOptions?.cookieJar,
+                } : overrideOptions?.cookieJar;
+
                 return gotScraping({
                     url: request!.url,
-                    method: request!.method,
+                    method: request!.method as Method, // Force cast to omit CONNECT
```

Comment:
its not "force casting", its type assertion :] i dont mind calling it casting (we all do, its convenient, although wrong), but "force casting" sounds quite weird

Reply:
You're right, I'll rename it